python/tests/io/test_pyarrow.py

```diff
@@ -1130,3 +1131,15 @@ def test_projection_filter_on_unknown_field(schema_int_str: Schema, file_int_str
         _ = project(schema, [file_int_str], GreaterThan("unknown_field", "1"), schema_int_str)
 
     assert "Could not find field with name unknown_field, case_sensitive=True" in str(exc_info.value)
+
+
+def test_pyarrow_to_schema_simple(table_schema_simple: Schema, pyarrow_schema_simple: pa.Schema) -> None:
+    actual = str(pyarrow_to_schema(pyarrow_schema_simple))
+    expected = str(table_schema_simple)
+    assert actual == expected
+
+
+def test_pyarrow_to_schema_nested(table_schema_nested: Schema, pyarrow_schema_nested: pa.Schema) -> None:
+    actual = str(pyarrow_to_schema(pyarrow_schema_nested))
+    expected = str(table_schema_nested)
```

Comment:
I used `str` instead of `repr` because `repr` exposes `schema_id=1,identifier_field_ids=[1]` which is not handled by the `pyarrow_to_schema` visitor and I think the visitor need not handle these since the schema generated by the visitor will not be put into any table metadata file.

Please correct me if I misunderstand something.